var/spack/repos/builtin/packages/py-pyfr/package.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+from spack import *
+
+
+class PyPyfr(PythonPackage):
+    """PyFR is an open-source Python based framework for solving
+    advection-diffusion type problems on streaming architectures
+    using the Flux Reconstruction approach of Huynh."""
+
+    homepage = "http://www.pyfr.org/"
+    pypi = "pyfr/pyfr-1.13.0.tar.gz"
+    git = "https://github.com/PyFR/PyFR/"
+    maintainers = ["michaellaufer"]
+
+    # git branches
+    version("develop", branch="develop")
+    version("master", branch="master")
+
+    # pypi releases
+    version(
+        "1.13.0",
+        sha256="ac6ecec738d4e23799ab8c50dea9bdbd7d37bc971bd33f22720c5a230b8e7b2f",
+    )
+
+    variant("metis", default=True, description="Metis for mesh partitioning")
+    variant("scotch", default=False, description="Scotch for mesh partitioning")
+    variant("cuda", default=False, description="CUDA backend support")
+    variant("hip", default=False, description="HIP backend support")
+
+    # Required dependencies
+    depends_on("python@3.9:", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
+    depends_on("py-gimmik@2.2:", type=('build', 'run'))
+    depends_on("py-h5py@2.10:", type=('build', 'run'))
+    depends_on("py-mako@1.0.0:", type=('build', 'run'))
+    depends_on("py-mpi4py@3.1.0:", type=('build', 'run'))
+    depends_on("py-numpy@1.20:+blas", type=('build', 'run'))
+    depends_on("py-platformdirs@2.2.0:", type=('build', 'run'))
+    depends_on("py-pytools@2016.2.1:", type=('build', 'run'))
+    depends_on("py-scipy", type=('build', 'run'))
+
+    # Optional  dependecies
+    depends_on("metis@5.0:", when="+metis", type=('build', 'run'))
+    depends_on("scotch@6.0:", when="+scotch", type=('build', 'run'))
+    depends_on("cuda@8.0:", when="+cuda", type=('build', 'run'))
+    depends_on("rocblas@4.5.0:", when="+hip", type=('build', 'run'))
```

Comment:
Not sure if these should be build/run or build/link (the default). build/run means to add `prefix.bin` to `PATH` during build-time and run-time, link means that the packages links to libraries/headers.

Reply:
Is there a specific package for HIP?

Reply:
@amd-toolchain-support might know the answer to this question

Reply:
@FreddieWitherden, there is a pacakge for HIP, but it gets pulled in implicitly by the rocblas package.

Reply:
After some testing, and inspection of the PyFR source code, it seems that PyFR uses these opportunistically at run time, such that ` type=('run'))` is all that is needed.